src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -687,9 +702,8 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 	public ensureDefaultLanguageFlavor(uri: string): void {
 		if (!this.getProviderIdFromUri(uri)) {
 			// Lookup the default settings and use this
-			let defaultProvider = WorkbenchUtils.getSqlConfigValue<string>(this._configurationService, Constants.defaultEngine);
-			if (defaultProvider && this._providers.has(defaultProvider)) {
-				// Only set a default if it's in the list of registered providers
+			let defaultProvider = this.getDefaultProviderId();
+			if (defaultProvider) {
```

Comment:
why are you removing this check ` && this._providers.has(defaultProvider)`?  It seems like a reasonable thing to validate here.

Reply:
The method getDefaultProviderId does the exact same thing, it does check for this._providers.has(defaultProvider). So just replacing with that.

Reply:
makes sense.  thanks for the clarification.